android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -892,11 +892,11 @@ public void loadNewContent() {
 
     @Override
     public void getFeed(boolean isNewContent) {
+        if (!isNewContent) {
+            mNtpAdapter.setImageCreditAlpha(1f);
+            mNtpAdapter.setNewsLoading(true);
+        }
         PostTask.postTask(TaskTraits.BEST_EFFORT_MAY_BLOCK, () -> {
-            if (!isNewContent) {
-                mNtpAdapter.setImageCreditAlpha(1f);
-                mNtpAdapter.setNewsLoading(true);
-            }
             initBraveNewsController();
             mBraveNewsController.getFeed(feed -> { runFeed(isNewContent, feed); });
```

Comment:
I'm not really sure that will work as `runFeed` has `mNtpAdapter.notifyItemRangeRemoved` inside, basically the same as the original problem inside `mNtpAdapter.setNewsLoading(true);`. I would probably revert the PR that fixes ANR and try to fix it different way, in particular by launching the rate dialog in a separate thread and not the whole feed re-init.

Reply:
@SergeyZhukovsky 
Correct, There are couple of other methods which called on mNtpAdapter object in this flow.
Those methods should call in UI thread.
But after this change when I verify crash is not happening.
ANR caused because all the things are happening in UI thread. 
Probably instead of reverting we can call all those methods which call on mNtpAdapter object in UI thread. that is where we are calling notify** methods.
Please let me know if it will works.